### PR TITLE
Added entity and migration

### PR DESCRIPTION
**Pull request by praneethpr**

### CODE REVIEW

File: src/Entity/InformationProduct.php [hunk 1]
```diff
@@ -139,54 +139,105 @@ public function getCreators(): ?string
...
+     */
     public function getExternalDoi(): ?string
     {
         return $this->externalDoi;
     }
 
+    /**
+     * Set the external DOI.
+     *
+     * @param string|null $externalDoi
```
Comment (praneethpr) [inline src/Entity/InformationProduct.php:192]: Missing empty line

File: src/Entity/InformationProduct.php [hunk 2]
```diff
@@ -139,54 +139,105 @@ public function getCreators(): ?string
...
+     */
+    public function isPublished(): ?bool
     {
         return $this->published;
     }
 
+    /**
+     * Set the published state of the Information Product.
+     *
+     * @param boolean $published
```
Comment (praneethpr) [inline src/Entity/InformationProduct.php:215]: Missing empty line

File: src/Entity/InformationProduct.php [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+     */
+    public function getTitle(): ?string
+    {
+        return $this->title;
+    }
+
+    /**
+     * Set the title for this Information Product.
+     *
+     * @param string|null $shortTitle The title for this Information Product.
```
Comment (praneethpr) [inline src/Entity/InformationProduct.php:121]: Incorrect variable name "$shortTitle" used

File: src/Entity/InformationProduct.php [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+     * Set the publisher of the Information Product.
+     *
+     * @param string $publisher
+     * @return self
+     */
+    public function setPublisher(string $publisher): self
+    {
+        $this->publisher = $publisher;
+
+        return $this;
```
Comment (praneethpr) [inline src/Entity/InformationProduct.php:176]: Do we need setters to return ?

File: src/Entity/InformationProduct.php [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+            $this->researchGroups[] = $researchGroup;
+        }
+
+        return $this;
+    }
+
+    /**
+     * Remove a Research Group from this Information Product.
+     *
+     * @param ResearchGroup $researchGroup
```
Comment (praneethpr) [inline src/Entity/InformationProduct.php:277]: Missing empty line

File: src/Entity/InformationProduct.php [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+     */
+    public function setExternalDoi(?string $externalDoi): self
+    {
+        $this->externalDoi = $externalDoi;
+
+        return $this;
+    }
+
+    /**
+     * Is the Information Producted published?
```
Comment (praneethpr) [inline src/Entity/InformationProduct.php:203]:
```suggestion
     * Is the Information Product published?
```

File: src/Entity/InformationProduct.php (hunk 4 above)
Comment (mickel1138) [reply src/Entity/InformationProduct.php:176]: The make command does it, we are not consistent in other classes. I kinda like it, but we can remote it?

Comment (praneethpr) [reply src/Entity/InformationProduct.php:176]: I think from now on, we can add returns to setters

**Review by mickel1138 — APPROVED** [review]

File: src/Entity/InformationProduct.php [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+    /**
+     * The publisher of the Information Product.
+     *
+     * @var string
+     *
+     * @ORM\Column(type="text")
+     *
+     * @Assert\NotBlank(
+     *     message="A publisher is required."
+     *
```
Comment (mickel1138) [inline src/Entity/InformationProduct.php:58]:
```suggestion
     *     message="A publisher is required."
     *  )
     *
```

**Review by fightingtexasaggie — APPROVED** [review]
We've got our start. ;-)

**Pull request merged**